src/verifier/mod.rs

```diff
@@ -119,12 +119,12 @@ impl Verifier {
         validator_function: F,
     ) -> Result<()>
     where
-        F: FnOnce(Session, AuthorizationResponse) -> Pin<Box<Fut>>,
-        Fut: Future<Output = Outcome>,
+        F: FnOnce(Session, AuthorizationResponse) -> Fut,
+        Fut: Future<Output = Result<Outcome>>,
```

Comment:
This is intentionally not `Result<Outcome>`. If there is an error, it should be captured in the `Outcome::Failure` variant.

Reply:
This still needs to be reverted.